### PR TITLE
+aws.0.1.0

### PR DESCRIPTION
**Pull request by seliopou**
This pull request replaces the existing aws client with the core library for a family of client libraries implemented via code generation. Separate libraries for each amazon web service will be published in a subsequent pull request.

/cc @barko

### CODE REVIEW

File: packages/aws/aws.0.1.0/url [hunk 1]
```diff
@@ -0,0 +1,2 @@
+url:"https://github.com/inhabitedtype/ocaml-aws/releases/download/aws-v0.1.0/aws-v0.1.0.tar.gz"
```
Comment (yallop) [inline packages/aws/aws.0.1.0/url]: This should be:

`archive: "https://github.com/inhabitedtype/ocaml-aws/releases/download/aws-v0.1.0/aws-v0.1.0.tar.gz"`

**Comment by seliopou** [comment]
Thanks, force-pushed the fix. I'll prepare another pull request with the service libraries shortly.

**Comment by yallop** [comment]
There's a [checksum mismatch](https://travis-ci.org/ocaml/opam-repository/jobs/108524435):

```
[ERROR] Bad checksum for
        /home/travis/.opam/packages.dev/aws.0.1.0/aws-v0.1.0.tar.gz:
          - 08ecc11ac94da76696d904f6e33fc7f4 [expected result]
          - 62f57266ba68237a82858472d030d6ad [actual result]
```

**Comment by seliopou** [comment]
nth time is a charm, hopefully.

**Comment by seliopou** [comment]
No idea what the current failure's about. It's only affects one build:

```
[ERROR] At ~/.opam/repo/default/packages/aws/aws.0.1.0/opam:9:53:
  Not a recognised version-control URL
```

File: packages/aws/aws.0.1.0/opam [hunk 2]
```diff
@@ -0,0 +1,47 @@
+opam-version: "1.2"
+name: "aws"
+version: "0.1.0"
+maintainer: "Spiros Eliopoulos <spiros@inhabitedtype.com>"
+authors: [ "Spiros Eliopoulos <spiros@inhabitedtype.com>"
+           "Daniel Patterson <dbp@dbpmail.net>" ]
+license: "BSD-3-clause"
+homepage: "https://github.com/inhabitedtype/ocaml-aws"
+dev-repo: "https://github.com/inhabitedtype/ocaml-aws"
```
Comment (dbuenzli) [inline packages/aws/aws.0.1.0/opam]: add `.git` here.

**Comment by dsheets** [comment]
Woohoo!

**Comment by barko** [comment]
I deleted my repository https://github.com/barko/aws to promote convergence .

**Comment by jj-issuu** [comment]
We were using the old aws.0.3.0 to access Amazon S3. Is there any way to access S3 with this new library? If not, shouldn't we let the old library stay in opam? It's fine to rename it to aws-legacy or something.

**Comment by seliopou** [comment]
The library that was just published only includes the core runtime library that other service bindings will use. Those libraries, including S3 will be published shortly (next couple days). In the mean time, you may want to add a version constraint to your opam files so you can continue to use `0.0.3`.

**Comment by agarwal** [comment]
@barko, @jj-issuu Note we'll also be releasing [oboto](https://github.com/solvuu/oboto) soon. In addition to the base AWS API, we have some higher level operations for S3. Those parts are intertwined in a private repo, but we're working to merge into oboto. Best.

**Pull request merged**